### PR TITLE
Query: Relational: Lift sub-queries in Where and OrderBy clauses.

### PR DESCRIPTION
**Pull request by anpete**

### CODE REVIEW

**Comment by anpete** [comment]
@maumar

File: src/EntityFramework.Relational/Query/Expressions/SelectExpression.cs [hunk 1]
```diff
@@ -44,6 +44,9 @@ public SelectExpression([NotNull] string alias)
             : base(null, Check.NotEmpty(alias, nameof(alias)))
         {
         }
+        public override Type Type => _projection.Count == 1
```
Comment (maumar) [inline src/EntityFramework.Relational/Query/Expressions/SelectExpression.cs]: add newline

File: src/EntityFramework.Core/Query/EntityQueryModelVisitor.cs [hunk 2]
```diff
@@ -490,11 +490,9 @@ var elementScoped
                         QueryResultScopeParameter),
                     Expression.Lambda(_expression, QueryResultScopeParameter));
 
-            var querySourceMapping = _queryCompilationContext.QuerySourceMapping;
-
-            if (!querySourceMapping.ContainsMapping(fromClause))
+            if (!_queryCompilationContext.QuerySourceMapping.ContainsMapping(fromClause))
```
Comment (maumar) [inline src/EntityFramework.Core/Query/EntityQueryModelVisitor.cs]: isn't the old version better? IIRC Resharper used to complain about scenarios like this.

**Comment by maumar** [comment]
:shipit:

**Pull request closed without merging**